gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -150,117 +201,300 @@ def test_unauthorized_user_cannot_list_tes_tasks(self):
         )
-        assert (
-            "id" in response
-        ), f"Response should contain a valid 'id' field. But found {response} instead"
-        task_id = response["id"]
 
-        response = self.gen3_workflow.list_tes_tasks(
+        task_id = task_response.get("id", None)
+        assert task_id, f"Expected 'id' in response, but got: {task_response}"
+
+        # Step 3: Verify task is listed
+        list_tes_tasks_response = self.gen3_workflow.list_tes_tasks(
             user=self.valid_user,
             expected_status=200,
         )
-        assert (
-            "tasks" in response
-            and isinstance(response["tasks"], list)
-            and len(response["tasks"]) > 0
-        ), f"The response should contain a non-empty list of tasks. But found {response} instead"
-
-        task_list = response["tasks"]
-        assert task_id in (
-            task["id"] for task in task_list
+
+        tasks_list = list_tes_tasks_response.get("tasks", [])
+        assert tasks_list, f"Expected non-empty list but got: {tasks_list}"
+
+        assert any(
+            task["id"] == task_id for task in tasks_list
         ), "The created task ID should be present in the task list."
 
-    def test_happy_path_get_tes_tasks(self):
-        """
-        Verify that an authorized user can retrieve a TES task by ID.
-        Expects: HTTP 200 and an id field.
-        """
-        response = self.gen3_workflow.create_tes_task(
-            user=self.valid_user,
-            expected_status=200,
-        )
+        # Step 4: Poll until the TES task completes or fails with a known status
+        max_retries = 10
+        poll_interval = 15  # seconds
+        transient_states = {"QUEUED", "INITIALIZING", "RUNNING"}
+        final_success_state = "COMPLETE"
+        final_failure_states = {"FAILED", "EXECUTOR_ERROR", "CANCELED"}
+
+        for attempt in range(1, max_retries + 1):
+            task_info = self.gen3_workflow.get_tes_task(
+                task_id=task_id,
+                user=self.valid_user,
+                expected_status=200,
+            )
+            state = task_info.get("state")
+
+            if state == final_success_state:
+                break
+            elif state in final_failure_states:
+                assert (
+                    False
+                ), f"TES task failed. Final state: {state}, Response: {task_info}"
+            elif state not in transient_states:
+                assert (
+                    False
+                ), f"Unexpected TES task state '{state}' encountered. Response: {task_info}"
+
+            logger.debug(f"Attempt {attempt}: Task state is '{state}', retrying...")
+            time.sleep(poll_interval)
+        else:
+            assert (
+                False
+            ), f"TES task did not complete in time. Final state: {state}, Response: {task_info}"
+
+        # Step 5: Validate task logs
+        stdout = task_info["logs"][0]["logs"][0]["stdout"].strip()
```

Comment:
this could break easily if the response does not have the expected format, could you add some checks or a try block?